empress/recon_vis/recon_viewer.py

```diff
@@ -1,80 +1,222 @@
+                        LineCollection([[(0, 0)]], linestyles=['dashed'],
+                                       colors=[LOSS_EDGE_COLOR], label='Loss')
+                      ]
+    fig.set_legend(legend_elements, title=consistency_type)
+
+
+def set_offsets(tree):
+    """
+    Populates the nodes of a Tree with an offset
+    :param tree: Tree Object
+    """
+    pos_dict = tree.pos_dict
+
+    for node in tree.postorder_list:
+        y_1 = None
+        y_0 = node.layout.row
+        for logical_pos in pos_dict:
+            if node.is_leaf:
+                if y_0 < logical_pos[0] and node.layout.col <= logical_pos[1]:
+                    if y_1 is None or y_1 > logical_pos[0]:
+                        y_1 = logical_pos[0]
+            else:
+                y_1 = max(node.left_node.layout.row, node.right_node.layout.row)
+        if y_1 is None or node.layout.node_count == 0:
+            node.layout.offset = TRACK_OFFSET
+        else:
+            node.layout.offset = abs(y_0 - y_1) / (node.layout.node_count + 3)
+
+
+def render_host(fig, host_tree, show_internal_labels, tip_font_size, internal_font_size):
+    """
+    Renders the host tree
+    :param host_tree: Host tree represented as a Tree object
+    :param show_internal_labels: Boolean that determines whether or not the internal labels are shown
+    :param tip_font_size: Font size for the text of the tips of the tree
+    :param internal_font_size: Font size for text of the internal nodes of the tree
+    """
     set_host_node_layout(host_tree)
     root = host_tree.root_node
     draw_host_handle(fig, root)
-    render_host_helper(fig, root, show_internal_labels)
+    render_host_helper(fig, root, show_internal_labels, tip_font_size, internal_font_size, host_tree)
 
 
 def draw_host_handle(fig, root):
-    """ Draw edge leading to root of host tree. """
+    """
+    Draw edge leading to root of host tree.
+    :param root: The root node of a tree object
+    """
     fig.line((0, root.layout.y), (root.layout.x, root.layout.y), HOST_EDGE_COLOR)
 
 
-def render_host_helper(fig, node, show_internal_labels):
-    """ Helper function for rendering the host tree. """
-    node_x, node_y = node.layout.x, node.layout.y
-    node_xy = (node_x, node_y)
-    if node.is_leaf():
-        fig.dot(node_xy)
-        fig.text(node_xy, node.name)
+def render_host_helper(fig, node, show_internal_labels, tip_font_size, internal_font_size, host_tree):
+    """
+    Helper function for rendering the host tree.
+    :param node: node object
+    :param show_internal_labels: Boolean that determines whether or not the internal labels are shown
+    :param tip_font_size: Font size for the text of the tips of the tree
+    :param internal_font_size: Font size for text of the internal nodes of the tree
+    :param host_tree: Tree object representing a Host Tree
+    """
+    host_tree.pos_dict[(node.layout.row, node.layout.col)] = node
+
+    Position = namedtuple('Position', ['x', 'y'])
```

Comment:
Please use one of the two syntaxes outline in the last comment as they are more readable. Also, move it to a different file.

Reply:
fixed